test_runner/regress/test_lsn_mapping.py

```diff
@@ -111,13 +108,56 @@ def test_lsn_mapping(neon_env_builder: NeonEnvBuilder):
         # make sure that we return the minimum lsn here at the start of the range
         assert Lsn(result["lsn"]) >= last_flush_lsn
 
+
+#
+# Test if cancelled pageserver get_lsn_by_timestamp request is correctly handled.
+#
+# Added as an effort to improve error handling and avoid full anyhow backtrace
+# https://github.com/neondatabase/neon/pull/7949
+#
+def test_get_lsn_by_timestamp_cancelled(neon_env_builder: NeonEnvBuilder):
+    env = neon_env_builder.init_start()
+
+    tenant_id, _ = env.neon_cli.create_tenant(
+        conf={
+            # disable default GC and compaction
+            "gc_period": "1000 m",
+            "compaction_period": "0 s",
+            "gc_horizon": f"{1024 ** 2}",
+            "checkpoint_distance": f"{1024 ** 2}",
+            "compaction_target_size": f"{1024 ** 2}",
+        }
+    )
+
+    timeline_id = env.neon_cli.create_branch(
+        "test_get_lsn_by_timestamp_cancelled", tenant_id=tenant_id
+    )
+    endpoint_main = env.endpoints.create_start(
+        "test_get_lsn_by_timestamp_cancelled", tenant_id=tenant_id
+    )
+    timeline_id = endpoint_main.safe_psql("show neon.timeline_id")[0][0]
```

Comment:
should be able to directly get an initial timeline ID from the env builder, but it's fine to copy other tests for now and fit it later.

Reply:
addressed in latest commit